src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Domain/LearnerCommitments.cs

```diff
@@ -14,65 +14,106 @@ public class LearnerCommitments
+            {
+                commitment.EffectiveStartDate = commitment.EffectiveFrom;
+
+                if (commitment.PaymentStatus == (int) PaymentStatus.Cancelled &&
+                    commitment.WithdrawnOnDate < commitment.EffectiveTo)
                 {
-                    if (commitment.WithdrawnOnDate.HasValue)
-                    {
-                        commitment.EffectiveEndDate = commitment.WithdrawnOnDate;
-                    }
+                    commitment.EffectiveEndDate = commitment.WithdrawnOnDate;
                 }
+                else
+                {
+                    commitment.EffectiveEndDate = commitment.EffectiveTo;
+                }
+
+                commitment.IsVersioned = true;
+            }
+        }
+
+        private static void SetupNonVersionedCommitment(Commitment commitment)
+        {
+            commitment.EffectiveStartDate = commitment.StartDate.LastDayOfMonth();
+            if (commitment.PaymentStatus == (int) PaymentStatus.Cancelled)
+            {
+                commitment.EffectiveEndDate = commitment.WithdrawnOnDate;
+            }
+            else
+            {
+                commitment.EffectiveEndDate = commitment.EndDate.LastDayOfMonth().AddDays(-1);
             }
+        }
 
+        private void AdjustLastCommitmentEndDateToInfinity()
+        {
             var lastCommitment = Commitments
                 .OrderByDescending(x => x.EffectiveStartDate)
                 .ThenByDescending(x => x.CommitmentId)
                 .FirstOrDefault();
-            if (lastCommitment != null)
+            if (lastCommitment != null && lastCommitment.PaymentStatus == (int) PaymentStatus.Active)
             {
-                lastCommitment.EffectiveEndDate = null;
+                lastCommitment.EffectiveEndDate = new DateTime(2099, 01, 01);
             }
         }
 
-        public IReadOnlyList<Commitment> ActiveCommitmentsForDate(DateTime date)
+        private void AdjustCommitmentEndDatesToMatchFollowingCommitmentsStartDate()
         {
-            return Commitments.Where(x => x.EffectiveStartDate <= date &&
+            for (var i = 0; i < Commitments.Count - 1; i++) // not looking at the last one
+            {
+                if (Commitments[i].EffectiveEndDate < Commitments[i + 1].EffectiveStartDate)
+                {
+                    Commitments[i].EffectiveEndDate = Commitments[i + 1].EffectiveStartDate.AddDays(-1);
+                }
+            }
+        }
 
-                                          (x.EffectiveEndDate == null ||
-                                          x.EffectiveEndDate >= date) &&
+        private void RemoveCommitmentsWithNoDays(IEnumerable<Commitment> commitmentList)
+        {
+            var commitmentsToRemove = new List<Commitment>();
+            foreach (var commitment in commitmentList)
+            {
+                if (commitment.EffectiveEndDate < commitment.EffectiveStartDate)
```

Comment:
Don't we want to remove commitment.EffectiveEndDate <= commitment.EffectiveStartDate.